packages/common/src/index.ts

```diff
@@ -345,18 +345,43 @@ export default class Common {
   }
 
   /**
-   * True if block number provided is the hardfork (given or set) change block of the current chain
+   * True if block number provided is the hardfork (given or set) change block
    * @param blockNumber Number of the block to check
    * @param hardfork Hardfork name, optional if HF set
    * @returns True if blockNumber is HF block
    */
   isHardforkBlock(blockNumber: number, hardfork?: string): boolean {
     hardfork = this._chooseHardfork(hardfork, false)
-    if (this.hardforkBlock(hardfork) === blockNumber) {
-      return true
-    } else {
-      return false
-    }
+    return this.hardforkBlock(hardfork) === blockNumber ? true : false
+  }
+
+  /**
+   * Returns the change block for the next hardfork after the hardfork provided or set
+   * @param hardfork Hardfork name, optional if HF set
+   * @returns Block number or null if not available
+   */
+  nextHardforkBlock(hardfork?: string): number | null {
+    hardfork = this._chooseHardfork(hardfork, false)
+    const hfBlock = this.hardforkBlock(hardfork)
+    // Next fork block number or null if none available
+    // Logic: if accumulator is still null and on the first occurence of
+    // a block greater than the current hfBlock set the accumulator,
+    // pass on the accumulator as the final result from this time on
+    const nextHfBlock = this.hardforks().reduce((acc: number, hf: any) => {
+      return hf.block > hfBlock && acc === null ? hf.block : acc
```

Comment:
This assumes that hardforks() returns a sorted array of forks on block numbers. This is logical, but is this also enforced? I see no checks that a `*.json` file in `chains/` needs to be sorted on increasing block numbers.

Reply:
Also, rinkeby has `dao` fork set to block number `null`, this would thus crash on this reduce call? (It is in the tests but in my mind this should crash?)

Reply:
Having a sorted array of fork blocks is not enforced but is assumed by convention and build upon all over in the library, so it shouldn't break into unchartered territory if assumed here as well. We can separately improve on this, not sure if this is necessary though. I think no one working on the library would add a new HF entry "just somewhere"? So I would say this is a very theoretical risk.

Reply:
`null > 1` (sample number) is just evaluated to `false` in that case, so the accumulator is left of `null` and goes in the next round.

Reply:
Woops, I assumed that `null > 3` would throw, but it does not.

Reply:
The "forks not in order" might be something we'd want to include if we make the library more safe, so for now it is fine by me.